File: C#/SRobotics/Core/Command/OnCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Command
{
    public class OnCommand : Commandable<State.ActivationStateType>
    {
        public void Execute(Robot context, State.ActivationStateType type)
        {
            context.State.ActivationState.Change(type);

            context.Commands.SayHello.Execute();
        }
    }
}

File: JAVA/SRobotics/src/sRobotics/core/command/OnCommand.java
package sRobotics.core.command;

import sRobotics.core.Robot;
import sRobotics.core.RobotContext;
import sRobotics.core.state.ActivationStateType;

public class OnCommand implements Commandable<ActivationStateType>
{
    public void Execute(RobotContext context, ActivationStateType type)
    {
        context.getState().activationState.Change(context, type);
    }
}
